lib/src/handle_access_token_response.dart

```diff
@@ -71,7 +71,7 @@ Credentials handleAccessTokenResponse(
   validate(parameters['token_type'].toLowerCase() == 'bearer',
       '"$tokenEndpoint": unknown token type "${parameters['token_type']}"');
 
-  var expiresIn = parameters['expires_in'];
+  var expiresIn = (parameters['expires_in'] is String) ? int.parse(parameters['expires_in']) : parameters['expires_in'];
```

Comment:
Should add on onError method to int.parse which calls the validate method if int.parse fails.